sheepdog/auth/__init__.py

```diff
@@ -19,6 +20,7 @@
 
 logger = get_logger(__name__)
 
+AUTHZ_CACHE = SimpleCache(default_timeout=1)
```

Comment:
Looks good. I'll just mention that i think we may be able to cache for a bit more than 1 second for sheepdog (how long exactly TBD...) and that maybe the cache expiration should be a configuration. No strong feelings about either.
cc @Avantol13

Reply:
The performance boost with even a 1-second cache is significant. However, with cache being longer than 1 second, I’m concerned about the cache persisting even after the JWT token expires. 

Regarding the timeout being configurable, I can add this change before merging the PR or defer it to the backlog and address it later if needed.